doc/constraint_causal_discovery.rst

```diff
@@ -0,0 +1,132 @@
+.. _constraint_causal_discovery:
+
+==================================
+Constraint-based causal discovery
+==================================
+.. automodule:: dodiscover.constraint
+    :no-members:
+    :no-inherited-members:
+.. currentmodule:: dodiscover
+
+The following are a set of methods intended for (non-parametric) structure learning
+of causal graphs (i.e. causal discovery) given observational and/or interventional data
+by checking constraints in the form of conditional independences (CI). At a high level
+all constraint-based causal discovery, tests CI statements, where we use different CI
+statistical tests to test the following null hypothesis:
+
+:math:`H_0: X \perp Y | Z` and :math:`H_A: X \not\perp Y | Z`
+
+For a given node ``X`` and ``Y`` in the underlying causal graph of interest, and
+a conditioning set, ``Z``. CI tests can have a variety of different assumptions
+that make one better than another in different settings and data assumptions.
+For more information on the CI tests themselves, see :ref:`conditional_independence`.
+
+Fundamental Assumptions of Constraint-Based Causal Discovery
+------------------------------------------------------------
+The fundamental assumptions of all algorithms in this section are the Markov property assumption
+and the causal faithfulness assumption :footcite:`Spirtes1993` and :footcite:`Pearl_causality_2009`.
+
+The Markov assumption states that all d-separation statements in the causal graph imply a
+corresponding CI statement in the data. This is a core-assumption that users from graphical modeling
+may be familiar with.
+
+On the other hand, the causal faithfulness assumption states that all
+CI statements in the data map to a d-separation statement. That is, there are no accidental
+CI that occur in the data, which are not represented by a d-separation statement in the underlying
+causal graph. The causal faithfulness assumption is a very problematic assumption because in practice
+one might have data that is very weakly dependent, such that a CI test under a specified :math:`\alpha`
+level would fail to reject the null hypothesis and conclude the variables in question are CI. In higher
```

Comment:
Meh.  So you get a sparser graph.  I'd say a denser graph is more probabilistic in causal inference.  You get edges where there is no causality.

Reply:
Are you saying that having no removal of edges is more problematic?